### PR TITLE
[MERGE] web,*: selection in list views

### PR DESCRIPTION
**Pull request by aab-odoo**
*account

This commit allows the user to apply actions to all records
matching the current domain in list views. It also ensures that
the user is aware of the set of records actually selected.

The number of selected records is now displayed in the control
panel, and the user can, with an additional click, select all
records matching the current domain instead of the ones of the
current page.

Task 2185145

### CODE REVIEW

File: addons/web/static/tests/helpers/mock_server.js [hunk 1]
```diff
@@ -1093,6 +1093,23 @@ var MockServer = Class.extend({
...
+     *
+     * @private
+     * @param {string} model
+     * @param {Array} args
+     * @param {Object} kwargs
+     * @param {integer} [kwargs.limit]
+     * @returns {integer[]}
+     */
+    _mockSearch: function (model, args, kwargs) {
+        let records = this._getRecords(model, args[0]).map(r => r.id);
```
Comment (ged-odoo) [inline addons/web/static/tests/helpers/mock_server.js]: for fun, here's a one-liner:
```js
return this._getRecords(model, args[0]).map(r => r.id).slice(0, kwargs.limit || Number.MAX_VALUE);
```

File: addons/web/static/src/js/views/basic/basic_controller.js [hunk 2]
```diff
@@ -242,7 +242,7 @@ var BasicController = AbstractController.extend(FieldManagerMixin, {
      * Archive the current selection
      *
      * @private
-     * @param {string[]} ids
+     * @param {int[]} ids
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/basic/basic_controller.js]: I personally use `number` now, because it's a valid Typescript type

**Comment by ged-odoo** [comment]
PR reviewed. It looks good to me. Well done

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-merge

**Comment by pparidans** [comment]
@robodoo retry

(waiting for this in mobile :wink: )

**Pull request closed without merging**